tests/foreman/api/test_subnet.py

```diff
@@ -21,528 +21,524 @@
 
 :Upstream: No
 """
+import re
+
 import pytest
 from nailgun import entities
 from requests.exceptions import HTTPError
 
 from robottelo.datafactory import gen_string
 from robottelo.datafactory import generate_strings_list
 from robottelo.datafactory import invalid_values_list
-from robottelo.test import APITestCase
+from robottelo.datafactory import parametrized
 
 
-class ParameterizedSubnetTestCase(APITestCase):
-    """Implements parametrized subnet tests in API"""
+@pytest.mark.tier1
+def test_positive_create_with_parameter():
+    """Subnet can be created along with parameters
 
-    @pytest.mark.tier1
-    def test_positive_create_with_parameter(self):
-        """Subnet can be created along with parameters
+    :id: ec581cb5-8c48-4b9c-b536-302c0b7ec30f
 
-        :id: ec581cb5-8c48-4b9c-b536-302c0b7ec30f
+    :steps: Create Subnet with parameter that has single key and single
+        value
 
-        :steps: Create Subnet with parameter that has single key and single
-            value
+    :expectedresults: The Subnet is created with parameter
+    """
+    parameter = [{'name': gen_string('alpha'), 'value': gen_string('alpha')}]
+    subnet = entities.Subnet(subnet_parameters_attributes=parameter).create()
+    assert subnet.subnet_parameters_attributes[0]['name'] == parameter[0]['name']
+    assert subnet.subnet_parameters_attributes[0]['value'] == parameter[0]['value']
+
+
+@pytest.mark.tier1
+@pytest.mark.parametrize('name', **parametrized(generate_strings_list()))
```

Comment:
Male sure you add docstring metadata stating that this test is parameterized.

Reply:
Male sure :D that is sexism :smile: :smiley: